openedx_events/learning/data.py

```diff
@@ -0,0 +1,138 @@
+"""
+Data attributes for events within the architecture subdomain `learning`.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+from datetime import datetime
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class UserNonPersonalData:
+    """
+    Attributes defined for Open edX user object based on non-PII data.
+
+    Arguments:
+        id (int): unique identifier for the Django User object.
+        is_active (bool): indicates whether the user is active.
+    """
+
+    id = attr.ib(type=int)
+    is_active = attr.ib(type=bool, default=True)
+
+
+@attr.s(frozen=True)
+class UserPersonalData:
+    """
+    Attributes defined for Open edX user object based on PII data.
+
+    Arguments:
+        username (str): username associated with the Open edX user.
+        email (str): email associated with the Open edX user.
+        name (str): email associated with the Open edX user's profile.
+    """
+
+    username = attr.ib(type=str)
+    email = attr.ib(type=str)
+    name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class UserData:
+    """
+    Attributes defined for Open edX user object.
+
+    Arguments:
+        user_non_pii (UserNonPersonalData): user's Personal Identifiable
+        Information.
+        user_pii (UserPersonalData): user's Non Personal Identifiable
+        Information.
+    """
+
+    user_non_pii = attr.ib(type=UserNonPersonalData)
```

Comment:
I'd like to keep the access to the user data a little less convoluted.
With this, to access the ID, one would need to call `user_data_obj.user_non_pii.id`

We could do something like:
```python
@attr.s(frozen=True)
class UserNonPiiData:
    id = attr.ib(type=int)
    is_active = attr.ib(type=bool)


@attr.s(frozen=True)
class UserPiiData:
    username = attr.ib(type=str)
    email = attr.ib(type=str)
    name = attr.ib(type=str, factory=str)


@attr.s(frozen=True)
class UserData(UserNonPersonalData):
    pii = attr.ib(type=UserPersonalData)


user_data_obj = UserData(
    pii=UserPiiData(
        username="course_enrollment.user.username",
        email="course_enrollment.user.email",
        name="course_enrollment.user.profile.name"
    ),
    id=1,
    is_active=True
)
```
So the access is more direct, but we still make a strong distinction for pii data.

Reply:
I had to remove the is_active default when testing on the interactive console